### ISSUE TITLE
[Bug]: (Laravel) Deleting video's caption not removing the vtt file

### ISSUE THREAD

**Issue by maktaa1995** [ToshY/BunnyNet-PHP#125]
### Package version(s) affected

4.0.1

### Description

We are doing something to update the video subtitles content, so we added a functionality to remove the captions for the selected language, it actually deletes it from the captions list in the player, but the `.vtt` file is not being removed. and when i try to add a new captions for the same language it uses the old one.

### How to reproduce

```php
$streamAPI->deleteCaption(libraryId: $libraryId, videoId: $bunny_video_id, sourceLanguage:  $language);
```
The api call is returning success response, but actually the video's subtitle file is still there.

### Possible Solution

Either add an optional param to `deleteCaption` method to force delete the `vtt` file or delete the file permanently.

**Comment by ToshY** [comment]
Hey @maktaa1995 👋 

Could you specify more details, like what you're actually using as `$language` value? I can image that the caption doesn't get deleted if you provide the incorrect language code.

> The api call is returning success response, but actually the video's subtitle file is still there.

Apparently, the response for this endpoint will always return 200 even if the subtitle with that language does not exist. I could add an additional note to README about this to prevent further confusion, but that's all I can do here. If you disagree that it returns 200 if the resource does not exists, you should create a support ticket at Bunny.net, as it's not the responsibility of this library.

> Either add an optional param to deleteCaption method to force delete the vtt file or delete the file permanently.

The `deleteCaption` is based on the [API specs](https://docs.bunny.net/reference/video_deletecaption), which does not have anything like "force delete".

I've tried to reproduce your problem  by adding subtitles (SRT/VTT) through the API (`addCaption`) and UI, en then perform a `deleteCaption`, which works fine on my end, as long as provide the same language code.

TLDR;

If your subtitle does not get deteled even though you think it should, please verify the language code. If it still fails, provide more details or (even better) a reproduction case.

**Comment by maktaa1995** [comment]
Hey @ToshY 

Thanks for your response and explanation, I did the same steps and followed all what you have mentioned in the docs.

I just found the issue, that bunny servers are cashing captions files so we have to purge the url of the video in order to have the new version of the caption file, and to perform this we have to use their base api to purge the url

`https://api.bunny.net/purge?url=https://{base-url}.b-cdn.net/{video-id}/*`

You can add it as a note in your docs so in case someone else faced the same issue.

**Comment by ToshY** [comment]
Hey @maktaa1995 👋 

> I just found the issue, that bunny servers are cashing captions files so we have to purge the url of the video in order to have the new version of the caption file, and to perform this we have to use their base api to purge the url

Do I understand it correctly that you've added a video and added a caption, then removed the caption and then checked the video again without purging it first, and saw it still using the old subtitles?

If that's the case, I think I understad why you were under the assumption the captions did not get deleted properly (and why you suggested something like "force delete").  FYI I was not aware that captions got cached separately.

After checking the [`Get Video Play Data`](**https://toshy.github.io/BunnyNet-PHP/stream-api/#get-video-play-data**) for a video, I see it returning a key with `captionsPath` containing something like this: `https://vz-abcd1234-fg5.b-cdn.net/d2cd04a8-12cc-4c3d-837f-9379bb995f0f/captions/`.

So my question to you is, instead of purging the entire video maybe you can also just purge the captions after you've updated them? If you could test/verify this, it would be much appreciated, and I could then update the docs with a sidenote for specifically purging the captions.

**Comment by maktaa1995** [comment]
Hey @ToshY 

Yeah exactly that what i was facing.

Also me didn't know that they are cashing the captions files, but it was the last try and it works 😄 .

Yeah i think it makes since to purge only captions link instead of the whole video link, will give it a try tomorrow an let you know.

And again thanks for your support!

**Comment by maktaa1995** [comment]
Hey @ToshY 

I did purge the link you sent yesterday, and yes it works, so no need to purge the whole video link only captions is enough.

Regards

**Comment by ToshY** [comment]
Hey @maktaa1995 

Thanks for the confirmation! I will create a PR to update the docs and close this issue after it is merged.
